### PR TITLE
TILA-1950 | Opening hours client enhancements

### PR DESCRIPTION
**Pull request by nicobav**
## Change log
- Adds new resource states to State enum in openinhours.enum; maintenance, not_in_use
- Add util method to State enum to return states that are to be considered when the reservations are possible
- Make the opening hours client know the usage of `end_time_on_next_day` value
- Make the opening hours client interpret resource to be reservable whole day when start_time and end_time are both nulls.

### CODE REVIEW

File: opening_hours/utils/opening_hours_client.py [hunk 1]
```diff
@@ -164,12 +168,20 @@ def get_resource_periods(self, resource) -> List[Period]:
...
-        for time in times:
-            if time.start_time <= start_time and time.end_time >= end_time:
+        for time in [
+            time
+            for time in times
+            if ResourceState(time.resource_state) in ResourceState.reservable_states()
+        ]:
+            if (
+                time.start_time <= start_time
+                and (time.end_time >= end_time or time.end_time_on_next_day)
```
Comment (saaste) [inline opening_hours/utils/opening_hours_client.py:182]: Hmm, I wonder if the first two lines should be in parenthesis 🤔 The first and last condition are both checking `time.start_time` but on the latter it is a falsy check and not comparison to some other value. So something like this:
```
(
    time.start_time <= start_time
    and (time.end_time >= end_time or time.end_time_on_next_day)
)
or (not time.start_time and not time.end_time)
```

Or maybe even other way round just to make the falsy check first 🤔

**Review by saaste — CHANGES_REQUESTED** [review]
Looks good! Just one question to make sure the condition check is correct. Also, tests seem to be failing.

File: opening_hours/utils/opening_hours_client.py (hunk 1 above)
Comment (nicobav) [reply opening_hours/utils/opening_hours_client.py:182]: nice ones :+1:  I'll refactor

Comment (nicobav) [reply opening_hours/utils/opening_hours_client.py:184]: Added the falsy check as variable to be more readable

**Review by saaste — APPROVED** [review]
Splendid! :shipit:

**Pull request merged**